opm/output/data/InterRegFlow.hpp

```diff
@@ -0,0 +1,361 @@
+/*
+  Copyright (c) 2022 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_OUTPUT_DATA_INTERREGFLOW_HPP
+#define OPM_OUTPUT_DATA_INTERREGFLOW_HPP
+
+#include <algorithm>
+#include <cassert>
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <functional>
+#include <iterator>
+#include <type_traits>
+#include <utility>
+
+namespace Opm { namespace data {
+
+    /// Intermediary Protocol to Linearise Per-Connection Flow Rates Into Subrange.
+    ///
+    /// \tparam RandIt Random access iterator type of subrange.
+    template <typename RandIt>
+    class InterRegFlow
+    {
+    public:
+        /// Subrange element type.
+        ///
+        /// Typically \c double or \c float.
+        using ElmT = std::remove_cv_t<
```

Comment:
This will be `double`- today, tomorrow and the day after.

Reply:
> This will be `double`&mdash;today, tomorrow and the day after.

In this case it's actually a [`float`](https://github.com/OPM/opm-common/blob/b5de771251819dd81b3f541105cf7690f4dcf865/opm/output/data/InterRegFlowMap.hpp#L57-L59).

Reply:
OK - point taken; or `float` ...